assets/app/scripts/services/auth.js

```diff
@@ -118,7 +118,7 @@ angular.module('openshiftConsole')
 
       requestRequiresAuth: function(config) {
         // TODO: replace with real implementation
-        var requiresAuth = config.url.toString().indexOf("api/") > 0;
+        var requiresAuth = (config.auth && config.auth.requiresAuth) || config.url.toString().indexOf("api/") > 0;
```

Comment:
@liggitt OK, the auth changes are in auth.js and data.js if you want to take a look. (The rest of the pull request is not ready for review yet.)